src/main.cpp

```diff
@@ -382,5 +407,51 @@ void print_help_message() {
 void print_version_information() {
   std::cout << "quick-lint-js version " << QUICK_LINT_JS_VERSION_STRING << '\n';
 }
+
+#if defined(_WIN32)
+std::vector<char *> wargv_to_mbargv(int argc, wchar_t **wargv) {
+  std::vector<char *> mbargv;
+  for (int i = 0; i < argc; i++) {
+    mbargv.emplace_back(wstring_to_mbstring(wargv[i]));
+  }
+  return mbargv;
+}
+
+char *wstring_to_mbstring(wchar_t *wstring) {
+  int size_required = WideCharToMultiByte(
+      /*CodePage=*/CP_UTF8,
+      /*dwFlags=*/0,
+      /*lpWideCharStr=*/wstring,
+      /*cchWideChar=*/-1,
+      /*lpMultiByteStr=*/nullptr,
+      /*cbMultiByte=*/0,
+      /*lpDefaultChar=*/NULL,
+      /*lpUsedDefaultChar=*/NULL);
+  if (size_required == 0) {
+    goto fail;
+  }
+  char *mbstring;
+  mbstring = new char[size_required];
```

Comment:
I'm pretty sure I have to allocate the string on the heap for mbargv. I don't like how I do a loop to de-allocate all the strings returned from this function. I think I read about smart pointers and how when a vector destructs it will call the destructor for all the things it is storing. Which means that the if I use smart pointers, I won't have to loop to de-allocate the string. Sounds like it's what I ought to use.

Reply:
> I'm pretty sure I have to allocate the string on the heap for mbargv.

Correct. `std::string` is probably a good option for that. A smart pointer like `std::unique_ptr<char>` would also work.